### ISSUE TITLE
🐛 BUG: blog-multiple-authors cannot be built: “MarkdownFrontmatter is not defined“

### ISSUE THREAD

**Issue by bramus** [withastro/astro#1551]
### What package manager are you using?

npm

### What operating system are you using?

Mac

### Describe the Bug

The `blog-multiple-authors` template cannot be built.

### Steps to Reproduce

1. Init + Try Building
  ```
  npm init astro -- --template blog-multiple-authors
  npm i
  npm run build
  ```
2. Get a build error:
  ```
  > @example/blog-multiple-authors@0.0.1 build
  > astro build
  
  [config] Set "buildOptions.site" to generate correct canonical URLs and sitemap
  [14:16:49] [snowpack] Hint: run "snowpack init" to create a project config file. Using defaults...
  [14:16:51] [snowpack] Ready!
  [build] ! building pages...
  [build] MarkdownFrontmatter is not defined
  ```

### Link to Minimal Reproducible Example (Optional)

https://github.com/snowpackjs/astro/tree/main/examples/blog-multiple-authors

**Comment by drwpow** [comment]
👋🏻 Hey Bramus!

> (The compiler didn't really warn me though 😅)

That’s good feedback. Agree that this should be more clear, and I thought we had a warning message at some point 🤔 (if not this should be easy to add).

Having `getStaticPaths()` execute before anything else is one of those “have-your-cake-and-eat-it-too” things—as it can do a lot of work and can supply props and everything else to the rest of the page, by nature it has to be isolated and execute first. If we tried to execute the entire scope as one, we’d get the opposite error, with some of your props being undefined because `getStaticPaths()` hasn’t fetched the data yet.

>Trying to work around this by …
>
>1. creating a separate file that contains the interface
>2. importing that interface
>
>… leads to another issue, as reported in #1046

Ah that’s also something we’re working on, as you pointed out. I will definitely add importing interfaces in our blog example! I agree it’s important.

**Comment by drwpow** [comment]
Update: this specific issue seems to have been fixed in `0.21` beta (`astro@next`). I know that’s still in beta and isn’t production-ready yet, but at least it will be fixed with the next release!

**Comment by jasikpark** [comment]
I can confirm that the complex blog starter example builds as of `v0.21.12`: https://stackblitz.com/github/mkliboprr?file=README.md

If there's another bug here, please comment and we can re-open, but it looks like that solves it!

Please create a discussion in https://github.com/withastro/rfcs/discussions/new if you think we need to add additional error reporting to make `getStaticPaths` easier to work with!
